Release/src/http/common/http_compression.cpp

```diff
@@ -156,7 +156,7 @@ class zlib_compressor_base : public compress_provider
 
 private:
     int m_state{Z_BUF_ERROR};
-    z_stream m_stream{0};
```

Comment:
What issue did this initializer cause?  And will this empty initializer ensure that all members of the z_stream struct are initialized to zero?  I **think** it might, but I'm not confident enough in my reading of the spec on that point to swear to it.

If not, then we'll need a memset to zero in the zlib_compressor_base (and zlib_decompressor_base) constructor.

Reply:
All the errors are listed in #887. This one in particular generated

```
Release/src/http/common/http_compression.cpp:159:24: error: missing field 'avail_in' initializer
      [-Werror,-Wmissing-field-initializers]
    z_stream m_stream{0};
```

I was surprised by this one as well. 

https://stackoverflow.com/questions/1538943/why-is-the-compiler-throwing-this-warning-missing-initializer-isnt-the-stru

Seems like it is, as the one poster in that SO put it, "GCC is just being overly paranoid". That posting is old, but yet it seems like the issue still exists. It looks like there have been requests to change it as well.

Let me know how you want this handled, in the meantime I'll make the other changes you listed above later today.

Reply:
Oh and sorry for not directly replying. They should be the same.

https://stackoverflow.com/questions/41006259/brace-initialization-difference-betweein-and-0-in-c

https://en.cppreference.com/w/cpp/language/value_initialization

Reply:
I fixed the two items you mentioned. I'm not going to mark this as resolved ... if you feel this is suitable, then you can mark it as such. Or let me know what you'd like to see.

Reply:
I'd like @BillyONeal to confirm -- he'll have to review this, anyway, as I don't have permission to merge it -- but the links are helpful.